internal/topology/check/compatibility_test.go

```diff
@@ -179,3 +181,84 @@ func TestCheckReferencedObjectsAreStrictlyCompatible(t *testing.T) {
 		})
 	}
 }
+
+func TestTemplatesAreCompatible(t *testing.T) {
+	template := clusterv1.LocalObjectTemplate{
+		Ref: &corev1.ObjectReference{
+			Namespace:  "default",
+			Name:       "foo",
+			Kind:       "bar",
+			APIVersion: "test.group.io/versionone",
+		},
+	}
+	compatibleNameChangeTemplate := clusterv1.LocalObjectTemplate{
+		Ref: &corev1.ObjectReference{
+			Namespace:  "default",
+			Name:       "foo",
+			Kind:       "bar",
+			APIVersion: "test.group.io/versionone",
+		},
+	}
+	compatibleAPIVersionChangeTemplate := clusterv1.LocalObjectTemplate{
+		Ref: &corev1.ObjectReference{
+			Namespace:  "default",
+			Name:       "foo",
+			Kind:       "bar",
+			APIVersion: "test.group.io/versiontwo",
+		},
+	}
+	incompatibleNamespaceChangeTemplate := clusterv1.LocalObjectTemplate{
+		Ref: &corev1.ObjectReference{
+			Namespace:  "different",
+			Name:       "foo",
+			Kind:       "bar",
+			APIVersion: "test.group.io/versionone",
+		},
+	}
+	incompatibleAPIGroupChangeTemplate := clusterv1.LocalObjectTemplate{
+		Ref: &corev1.ObjectReference{
+			Namespace:  "default",
+			Name:       "foo",
+			Kind:       "bar",
+			APIVersion: "production.group.io/versionone",
+		},
+	}
+	tests := []struct {
+		name    string
+		current clusterv1.LocalObjectTemplate
+		desired clusterv1.LocalObjectTemplate
+		wantErr bool
+	}{
+		{
+			name:    "Allow change to template name",
+			current: template,
+			desired: compatibleNameChangeTemplate,
```

Comment:
nit: Might make sense to inline all the templates except `template`. Currently it almost feels like two tables.

Absolutely fine for me to keep as is, if you prefer it.

Reply:
I prefer it as is, and I think the test is simple enough that the var names should be enough to understand what's going on.